platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -523,17 +526,22 @@ private void logout() {
   }
 
   private void logSecurityAuditRole() {
+    SecurityToken IdpSecToken = getIdpSecurityToken();
     if (shouldAuditSubject()) {
       SecurityLogger.audit(
           "Subject with admin privileges has logged out: {}",
-          getIdpSecurityAssertion().getPrincipal().getName());
+          new SecurityAssertionImpl(IdpSecToken).getPrincipal().getName());
     }
   }
 
   private boolean shouldAuditSubject() {
+    SecurityToken IdpSecToken = getIdpSecurityToken();
```

Comment:
✏️ This is technically unrelated to your PR, but thinking we should have the `logSecurityAuditRole` method just pass the token into this method since that's the only usage of this method. Otherwise, we're getting the token twice in a row.